src/Compilers/Core/Portable/Emit/Context.cs

```diff
@@ -2,32 +2,33 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
-#nullable disable
-
 using System;
 using System.Diagnostics;
 
 namespace Microsoft.CodeAnalysis.Emit
 {
-    internal struct EmitContext
+    internal readonly struct EmitContext
     {
         public readonly CommonPEModuleBuilder Module;
-        public readonly SyntaxNode SyntaxNodeOpt;
+        public readonly SyntaxNode? SyntaxNodeOpt;
+        public readonly RebuildData? RebuildDataOpt;
```

Comment:
This is one part of the PR I'm less sure on. This seems like an appropriate bag to put the `RebuildData` into as it's relevant to emit. There aren't comments here so I'm not sure if there invariants around this which having `RebuildData` inside violates.

Reply:
Last time I worked in this space I remember it wasn't clear to me what belongs to EmitContext and what belongs to CommonPEModuleBuilder. Seems like we could potentially merge these... but IDK.

Reply:
Nit: do we need -Opt suffix now what the type is annotated?

Reply:
We tend to remove such suffixes on reference types when nullable-enabling code.

Reply:
I can remove the `Opt` suffix here. Didn't originally bc wanted to keep the diff a bit cleaner and not introduce unnecessary noise (particularly if we weren't going to agree this was the right place). Will fix this up.

Reply:
You all asked for it 😄 

https://github.com/dotnet/roslyn/pull/52960/commits/cd8b78f135e0862d4154c5aa49539e2ec29bd713

Only took VS a few minutes to chew through the rename.